### PR TITLE
Add syntax highlighting to the YamlView

### LINKED ISSUE #3513: Add syntax highlighting to YamlView
If we can colorize the keys in the YamlView `yaml` string prop, dense yaml would be a lot easier to parse through. 

AC:
- Is it possible?
- keys become `primary10`

@JamWils :

> have the YAML color code keys and values

### PR DESCRIPTION
**Pull request by opudrovs**
Closes #3513

Changes:
- Added syntax highlighting to the YamlView with `react-syntax-highlighter`.
- Added support for light and dark modes in the YAML view (with passing the mode prop for now).
- Updated snapshots.

Notes:
- When copying YAML directly from browser (from the highlighted view) — after enabled the option to wrap long lines in the view — newlines are inserted into each line after key and after colon because of `display: flex` added to lines, so in Chrome it's copied like that:
```
id
:
value
```
In the original unstyled view no newlines were inserted at all, so the copied YAML was not formatted properly and was copied like `id1:    value1  id2:    value2`.

So, copying formatted YAML directly from browser does not look feasible without using a text component.

Testing:

To test the light mode, view YAML without passing any additional props to the `YamlView` component. To test the dark mode, pass the `UiMode.Dark` as the mode prop to `YamlView`.

Screenshots in light and dark mode:

Light mode:

Screenshot 2023-05-08 at 11 52 24

Screenshot 2023-05-08 at 11 52 35

Screenshot 2023-05-08 at 11 52 47

Dark mode:

Screenshot 2023-05-08 at 12 17 28

Screenshot 2023-05-08 at 12 17 38

Screenshot 2023-05-08 at 12 18 35

### CODE REVIEW

**Comment by opudrovs** [comment]
Also pulled this branch into enterprise and tested it there:

Screenshot 2023-05-08 at 14 33 53

Screenshot 2023-05-08 at 14 34 10

File: ui/components/YamlView.tsx [hunk 1]
```diff
@@ -38,11 +64,15 @@ function UnstyledYamlView({ yaml, object, className }: YamlViewProps) {
...
-      <pre>
-        {yaml?.split("\n").map((yaml, index) => (
-          <code key={index}>{yaml}</code>
-        ))}
-      </pre>
+      <SyntaxHighlighter
+        language="yaml"
+        {...styleProps}
+        wrapLongLines={true}
+        wrapLines={true}
```
Comment (jpellizzari) [inline ui/components/YamlView.tsx:71]:
```suggestion
        wrapLines
```

File: ui/components/YamlView.tsx [hunk 2]
```diff
@@ -38,11 +64,15 @@ function UnstyledYamlView({ yaml, object, className }: YamlViewProps) {
...
       </YamlHeader>
-      <pre>
-        {yaml?.split("\n").map((yaml, index) => (
-          <code key={index}>{yaml}</code>
-        ))}
-      </pre>
+      <SyntaxHighlighter
+        language="yaml"
+        {...styleProps}
+        wrapLongLines={true}
```
[the remark this answers is not in the document]
Comment (opudrovs) [reply ui/components/YamlView.tsx]: Replaced.

File: ui/components/YamlView.tsx (hunk 1 above)
Comment (opudrovs) [reply ui/components/YamlView.tsx]: Replaced.

**Pull request merged**